Sources/Promises/Promise+Do.swift

```diff
@@ -27,7 +27,7 @@ public extension Promise {
     let objCPromise = ObjCPromise<AnyObject>.__onQueue(queue) {
       do {
         let resolution = try work()
-        return resolution as? NSError ?? Promise<Value>.asAnyObject(resolution)
+          return type(of: resolution) is NSError.Type ? resolution as! NSError : Promise<Value>.asAnyObject(resolution)
```

Comment:
Let's use this formatting (convention for this codebase):
```
return type(of: resolution) is NSError.Type
  ? resolution as! NSError : Promise<Value>.asAnyObject(resolution)
```

Also applies to the change in Promise+Then